### ISSUE TITLE
Temperature sensor not considered in ClimateGetTemperature

### ISSUE THREAD

**Issue by dontinelli** [home-assistant/intents#1796]
While trying out the newly translated intents I found out that my temperature sensors are not considered. I get the following error:
`raise intent.IntentHandleError("No climate entities")`
I thought that in prior versions I did get a result.
I understand that the temperature sensor does not qualify as climate device, insofar it makes sense not to consider these sensors in the ClimateGetTemperature intents. I guess the temperature sensor should then be handled over HassGetState. In my view this has two downsides, though: First, the same sentences have to be doubled, which could lead to conflicts. Second, if a user asks about the temperature of an area, it seems not predictable, if the temperature of a temperature sensor or a climate device is returned.
What are your thoughts on this?

**Comment by tetele** [comment]
First of all, the intent is called `ClimateGetTemperature`, with an emphasis on `Climate`. It's meant to work for `climate` entities, which don't even expose the temperature in the `state`, but in an attribute.

Second,
> I thought that in prior versions I did get a result.
You still do if you ask `what is [the temperature of] <temperature sensor name or alias>?`

Third, we have long considered allowing multiple "filters" for a single sentence, to select entities in different domains, e.g.
- `{domain: "climate"} + {domain: "sensor", device_class: "temperature"}` for sentences such as `what is the temp in the living room?`
- `{domain: "cover", device_class: "door"} + {domain: "binary_sensor", device_class: "door"}` for sentences such as `how many doors are open?`
but the implementation was horrible and never got around to it.

Still, even if we do it, there will be 2 issues to tackle:
- a `climate` entity most likely shows the ambiental temperature. A temp `sensor` may be that of a 3D printer
- how should you tackle multiple readings in a single `area`? First found entity? Average? Min-max?

To sum up: we'll get to that, but not yet.

**Comment by dontinelli** [comment]
Thank you @tetele for your explanations. Makes totally sense to me.

**Comment by tetele** [comment]
Also, this https://github.com/home-assistant/architecture/discussions/1017
